src/ast-utils/ast-value.ts

```diff
@@ -116,45 +121,38 @@ function binaryExpressionToValue(member) {
 /**
  * MemberExpression references a variable with name
  */
-function memberExpressionToValue(member) {
+function memberExpressionToValue(member:ESTree.MemberExpression) {
   return expressionToValue(member.object) + "." + expressionToValue(member.property);
 }
 
 /**
- * Tries to get a value from expression. Handles Literal, UnaryExpression
- * returns undefined on failure
- * valueExpression example:
- * { type: "Literal",
+ * Tries to get the value of an expression. Returns undefined on failure.
  */
-function expressionToValue(valueExpression) {
+export function expressionToValue(valueExpression:ESTree.Node):LiteralValue {
   switch(valueExpression.type) {
     case 'Literal':
-      return literalToValue(valueExpression);
+      return literalToValue(<ESTree.Literal>valueExpression);
```

Comment:
I thought TS has a way to infer the types here. Can you add the value of Node.type to the interfaces for the node classes so that the value check on .type acts as a type guard?